simpleflow/executor.py

```diff
@@ -118,3 +111,24 @@ def fail(self, reason, details=None):
 
         """
         pass
+
+    def before_replay(self):
+        return self._workflow.before_replay(self._history)
+
+    def after_replay(self):
+        return self._workflow.after_replay(self._history)
+
+    def after_finished(self):
+        return self._workflow.after_finished(self._history)
+
+    def after_run(self):
+        """
```

Comment:
`warnings.warn("deprecated", PendingDeprecationWarning)`? (brownie points: as a decorator, changing the filter level, etc).